File: 06 - Branch-Sums/06.js
class BinaryTree {
    constructor(value) {
      this.value = value;
      this.left = null;
      this.right = null;
    }
  }

function branchSums(root){
    sums = []
    branchSumsHelper(root, sums, 0)
    return sums
}


function branchSumsHelper(root, sums, counter){
    if (root === null) return

    counter += root.value
    if (root.right === null && root.left === null) {
        sums.push(counter)
        return
    }
    branchSumsHelper(root.left, sums, counter)
    branchSumsHelper(root.right, sums, counter)
}

File: easy/06 - Branch-Sums/06.py
class BinaryTree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def branchSums(root):
    sums = []
    branch_sums_helper(root, sums, 0)
    return sums


def branch_sums_helper(root, sums, counter):
    if root is None:
        return
    else:
        counter += root.value
        if root.right is None and root.left is None:
            sums.append(counter)
            return
        branch_sums_helper(root.left, sums, counter)
        branch_sums_helper(root.right, sums, counter)
